exchanges/lbank/lbank_wrapper.go

```diff
@@ -718,7 +734,92 @@ func (l *Lbank) ValidateCredentials() error {
 	return l.CheckTransientError(err)
 }
 
+// FormatExchangeKlineInterval returns Interval to exchange formatted string
+func (l *Lbank) FormatExchangeKlineInterval(in kline.Interval) string {
+	switch in {
+	case kline.OneMin, kline.ThreeMin,
+		kline.FiveMin, kline.FifteenMin, kline.ThirtyMin:
+		return "minute" + in.Short()[:len(in.Short())-1]
+	case kline.OneDay:
+		return "day1"
+	case kline.OneWeek:
+		return "week1"
+	}
+	return in.Short()
+}
+
 // GetHistoricCandles returns candles between a time period for a set time interval
-func (l *Lbank) GetHistoricCandles(pair currency.Pair, a asset.Item, start, end time.Time, interval time.Duration) (kline.Item, error) {
-	return kline.Item{}, common.ErrFunctionNotSupported
+func (l *Lbank) GetHistoricCandles(pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	if !l.KlineIntervalEnabled(interval) {
+		return kline.Item{}, kline.ErrorKline{
+			Interval: interval,
+		}
+	}
+
+	data, err := l.GetKlines(l.FormatExchangeCurrency(pair, a).String(),
+		"2880", l.FormatExchangeKlineInterval(interval),
+		strconv.FormatInt(start.Unix(), 10))
+	if err != nil {
+		return kline.Item{}, err
+	}
+
+	ret := kline.Item{
+		Exchange: l.Name,
+		Pair:     pair,
+		Asset:    a,
+		Interval: interval,
+	}
+
+	for x := range data {
+		ret.Candles = append(ret.Candles, kline.Candle{
+			Time:   time.Unix(data[x].TimeStamp, 0),
+			Open:   data[x].OpenPrice,
+			High:   data[x].HigestPrice,
+			Low:    data[x].LowestPrice,
+			Close:  data[x].ClosePrice,
+			Volume: data[x].TradingVolume,
+		})
+	}
+
+	return ret, nil
+}
+
+// GetHistoricCandlesEx returns candles between a time period for a set time interval
+func (l *Lbank) GetHistoricCandlesEx(pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
```

Comment:
This does not return data to me using `btc_usdt, spot, time.Now().Add(-time.Hour), time.Now(), kline.OneMin` but `GetHistoricCandles` does. I couldn't see anything in particular that would prevent it.

Reply:
Fixed in upcoming commit! was using wrong index for iteration